Onboarding flows: SB page improvements, timeouts removal and other minor fixes

micieslak opened the pull request:
### What does the PR do

- exposes states controlling the flow to the Storybook page
- makes the Storybook page easier to use by adding contextual actions depending on the current page
- fixes some missing interactions with the store
- removes some timeouts that should not be handled directly on the ui level
- makes unit test for OnboardingLayout very fast

Closes: #17030
Closes: #17031
Closes: #17032

### Affected areas
Onboarding flows

### Architecture compliance

- [x] I am familiar with the application architecture and agreed good practices.
My PR is consistent with this document: [Status Desktop Architecture Guide](https://github.com/status-im/status-desktop/blob/master/CONTRIBUTING.md)

### Screenshot of functionality (including design for comparison)

- [x] I've checked the design and this PR matches it

<!-- screenshot (or gif/video) that demonstrates the functionality, specially important if it's a bug fix. -->

<!-- Uncomment this section for status-go upgrade/dogfooding pull requests

### Impact on end user

What is the impact of these changes on the end user (before/after behaviour)

### How to test

- How should one proceed with testing this PR.
- What kind of user flows should be checked?

### Risk 

Described potential risks and worst case scenarios.

Tick **one**:
- [ ] Low risk: 2 devs MUST perform testing as specified above and attach their results as comments to this PR **before** merging.
- [ ] High risk: QA team MUST perform additional testing in the specified affected areas **before** merging.

-->

status-im-auto commented:
### Jenkins Builds
| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | 015bf15b | [#1](https://ci.status.im/job/status-desktop/job/prs/job/linux/job/x86_64/job/tests-nim/job/PR-17048/1/) | 2025-01-09 22:11:30 | ~8 min | `tests/nim` | [:page_facing_up:`log`](https://ci.status.im/job/status-desktop/job/prs/job/linux/job/x86_64/job/tests-nim/job/PR-17048/1//consoleText) |

micieslak commented on ui/app/AppLayouts/Onboarding2/pages/KeycardCreatePinPage.qml:
```diff
@@ -103,7 +102,7 @@ KeycardBasePage {
             StateChangeScript {
                 script: {
                     pinInput.setPin(d.pin)
-                    Backpressure.debounce(root, 2000, () => root.keycardPinCreated(d.pin))()
```
yes, the behavior is the same

micieslak replied:
It's just moved up one level and parameterized, now the flow manages the delay, not page itself.

micieslak commented on storybook/pages/OnboardingLayoutPage.qml:
```diff
@@ -75,22 +79,25 @@ SplitView {
                 logs.logEvent("OnboardingStore.removeMnemonic()")
             }
 
-            readonly property int syncState: Onboarding.SyncState.InProgress // enum Onboarding.SyncState
```
it's just moved up, to group together with other properties representing states

micieslak commented:
> Nice work overall! My only doubt is about those timeouts and the PIN pages still working fine

@caybro I addressed your comments and added one more commit (externalizing metrics, as you suggested in the comment in code). Please take a look once again.

caybro commented on ui/app/AppLayouts/Onboarding2/OnboardingLayout.qml:
```diff
@@ -148,22 +140,13 @@ Page {
-                        "usage_data_shared",
-                        { placement: Constants.metricsEnablePlacement.onboarding })
-            localAppSettings.metricsPopupSeen = true
-        }
```
Good move imo; this is mainly for @jrainville, I guess you'll want to handle this signal  yourself from now in main.qml

caybro commented on ui/app/AppLayouts/Onboarding2/OnboardingLayout.qml:
```diff
@@ -9,30 +9,24 @@ import AppLayouts.Onboarding2.pages 1.0
 
     readonly property alias stack: stack
 
+    signal shareUsageDataRequested(bool enabled)
```
This is the one to hook into @jrainville

caybro reviewed (approved):
Nice!
